pkg/util/uuid_generator.go

```diff
@@ -33,3 +35,29 @@ func GenerateUuidFromName(name string) (string, error) {
 	uuid := uuidLib.NewMD5(namespaceUuid, []byte(name)).String()
 	return uuid, nil
 }
+
+// IsUuid tests whether a potential configId is already a UUID
+func IsUuid(configId string) bool {
+	_, err := uuidLib.Parse(configId)
+	if err == nil {
+		return true
+	} else {
+		return false
+	}
+}
+
+// GenerateUuidFromConfigId takes the unique project identifier within an environment, a config id and
+// generates a valid UUID based on provided information
+func GenerateUuidFromConfigId(projectUniqueId string, configId string) (string, error) {
+	// Return if configId is UUID
+	isUuid := IsUuid(configId)
+	if isUuid {
```

Comment:
I am not a fan of this method either returning OR generating a UUID, especially with the name and documentation not telling me! (and it being perfectly valid to generate uuids from uuids)
I've actually stumbled to understand the download/parse logic due to this behaviour making the downloaded non-unique-name ids work. 

I would prefer using this function only for generating UUIDs and doing the `IsUuid` check explicitly where needed. 
If my understanding/comment on` yaml_creator.go/parseConfigDetails` line 218 is correct I think the only place you will have to do the check and generate a UUID if it is not, is on deployment

Reply:
I think there's a misunderstanding. A UUID is always generated. In case of a download, the yaml file's configId (+ environment + project) is translated into a UUID. This UUID is matched with whatever UUID is returned from Dynatrace. With this information, Monaco knows which config instance to update.